pika/adapters/blocking_connection.py

```diff
@@ -1172,6 +1172,15 @@ def __int__(self):
     def __repr__(self):
         return '<%s impl=%r>' % (self.__class__.__name__, self._impl)
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, value, traceback):
+        try:
+            self.close()
+        except (exceptions.ChannelClosed, exceptions.ChannelAlreadyClosing):
```

Comment:
BlockingConnection is synchronous, so I wouldn't expect that it should ever see `exceptions.ChannelAlreadyClosing` percolating to its level. How are you able to reproduce `exceptions.ChannelAlreadyClosing` here?